app/app/wyoming.ts

```diff
@@ -712,13 +721,59 @@ class WyomingServer_ {
     this._connectionStateCallback?.(s);
   };
 
+  _increaseVolume16BitPCM(data: Uint8Array, gain: number) {
+    if (!(data instanceof Uint8Array)) {
+      throw new Error('Input must be a Uint8Array.');
+    }
+    if (data.byteLength % 2 !== 0) {
+      throw new Error('Uint8Array byteLength must be even for 16-bit samples.');
+    }
+    if (typeof gain !== 'number' || gain < 0) {
+      throw new Error('Gain must be a non-negative number.');
+    }
+
+    // Create a DataView to read and write 16-bit integers
+    const dataView = new DataView(data.buffer);
+    const numSamples = data.byteLength / 2;
+
+    // Create a new Uint8Array for the output
+    const outputUint8Array = new Uint8Array(data.byteLength);
+    const outputDataView = new DataView(outputUint8Array.buffer);
+
+    const MAX_16_BIT = 32767;
+    const MIN_16_BIT = -32768;
+
+    for (let i = 0; i < numSamples; i++) {
+      // Read the original 16-bit sample (little-endian)
+      const sample = dataView.getInt16(i * 2, true); // true for little-endian
+
+      // Apply gain
+      let newSample = Math.round(sample * gain * (gain * 0.6)); // roughly increasing curve from 0 to 92x
```

Comment:
have you found that a linear scale makes sense in practice? or would a [reverse-?]log scale feel more natural for users?

Reply:
This is to give a more logarithimic scale.  I think some devices will benefit from a small gain, say 1-5x (setting 1 to 2) whereas some need a much bigger gain (like my Lenovo LSD) at 60-80x (8-10).